ecscollector/collector.go

```diff
@@ -193,22 +210,41 @@ func (c *collector) Collect(ch chan<- prometheus.Metric) {
 	if metadata.Limits != nil {
 		if metadata.Limits.CPU != nil {
 			ch <- prometheus.MustNewConstMetric(
-				svcCPULimitDesc,
+				taskCpuLimitDesc,
 				prometheus.GaugeValue,
 				*metadata.Limits.CPU,
-				metadata.TaskARN,
 			)
 		}
 		if metadata.Limits.Memory != nil {
 			ch <- prometheus.MustNewConstMetric(
-				svcMemLimitDesc,
+				taskMemLimitDesc,
 				prometheus.GaugeValue,
-				float64(*metadata.Limits.Memory),
-				metadata.TaskARN,
+				float64(*metadata.Limits.Memory*mebibytes),
 			)
 		}
 	}
 
+	if metadata.EphemeralStorageMetrics != nil {
+		ch <- prometheus.MustNewConstMetric(
+			taskEphemeralStorageUsedDesc,
+			prometheus.GaugeValue,
+			float64(metadata.EphemeralStorageMetrics.UtilizedMiBs*mebibytes),
+		)
+		ch <- prometheus.MustNewConstMetric(
+			taskEphemeralStorageAllocatedDesc,
+			prometheus.GaugeValue,
+			float64(metadata.EphemeralStorageMetrics.ReservedMiBs*mebibytes),
+		)
+	}
+
+	if metadata.PullStartedAt != nil && metadata.PullStoppedAt != nil {
+		ch <- prometheus.MustNewConstMetric(
+			taskImagePullDurationDesc,
+			prometheus.GaugeValue,
+			float64(metadata.PullStoppedAt.Sub(*metadata.PullStartedAt))*nanoseconds,
```

Comment:
One Prometheus style thing. We typically want to expose underlying timestamps rather that compute durations like this.

If this is two metrics, `ecs_task_image_pull_start_timestamp_seconds` and `ecs_task_image_pull_stop_timestamp_seconds` you can do other useful calculations like `time() - ecs_task_image_pull_stop_timestamp_seconds`. to find out how long ago the pull stopped.

Reply:
They already do exist, in a weird sense, as labels on the info metric (with formatted timestamp values, not numeric values). But I'm not sure you can do these kinds of calculations with those labels. I'm not sure why they're on the info metric at all, to be honest.

What you're saying makes sense, I will expose these as separate gauges.

Reply:
Yes, those should be removed from the labels. You can't do math on label values.